### PR TITLE
✨ (feat): add checksum verification before upload

### LINKED ISSUE #309: Before upload, check if the checksum is the same and block the user if that the same
This is made to stop mistakes
We could also propose to user when upload to build for them

### PR DESCRIPTION
**Pull request by asr2003**
fixes #309 
/claim #309

### CODE REVIEW

**Comment by asr2003** [comment]
@riderx I am not sure what could be expected checksum. I felt lack of more issue description. If you suggest for the same, I am will incorporate them to satisfy the issue requirements.

I am more interested to complete this issue @riderx :)

**Comment by asr2003** [comment]
@riderx Can you also have a look at here?

**Comment by riderx** [comment]
You can use the same logic of **checkCompatibility** and instead of getting the native_packages of a version, you can get its checksum.
Same as **checkCompatibility** this requires an option to disable it.
you can do it just after verifyCompatibility

**Comment by asr2003** [comment]
Sure @riderx will update the same

File: src/utils.ts [hunk 1]
```diff
@@ -1019,3 +1019,19 @@ export async function checkCompatibility(supabase: SupabaseClient<Database>, app
     localDependencies: dependenciesObject,
   }
 }
+
+// Function to get the remote checksum
+export async function getRemoteChecksum(supabase: SupabaseClient<Database>, appId: string, channel: string) {
```
Comment (asr2003) [inline src/utils.ts:1024]: @riderx I am not sure is this the way of getting the checksum of remote same of logic of **checkCompatibility**. any suggestions on this?

File: src/bundle/upload.ts [hunk 2]
```diff
@@ -301,223 +302,248 @@ async function uploadBundleToCapgoCloud(supabase: SupabaseType, appid: string, b
   spinner.start(`Uploading Bundle`)
   const startTime = performance.now()
 
-  try {
-    if (options.multipart !== undefined && options.multipart) {
-      p.log.info(`Uploading bundle as multipart`)
```
Comment (riderx) [inline src/bundle/upload.ts:306]: Why all this code is deleted ?

File: src/utils.ts (hunk 1 above)
Comment (riderx) [inline src/utils.ts:1024]: This cannot work, you have the data model in Supabase types.
You need to test locally your code by running our backend or create an account in prod.

File: src/bundle/upload.ts [hunk 3]
```diff
@@ -34,6 +34,7 @@ interface Options extends OptionsBase {
   ignoreMetadataCheck?: boolean
   timeout?: number
   multipart?: boolean
+  disableChecksumVerification?: boolean;
```
Comment (riderx) [inline src/bundle/upload.ts:37]: You didn't test this either, the options are not added to the index file.
Honestly, I feel you make me lose my time here

File: src/utils.ts (hunk 1 above)
Comment (riderx) [reply src/utils.ts:1024]: This cannot work, you have the data model in Supabase types.
You need to test locally your code by running our backend or create an account in prod.

File: src/bundle/upload.ts (hunk 2 above)
Comment (asr2003) [reply src/bundle/upload.ts:306]: They aren't deleted. They moved to down in the changes :)

File: src/bundle/upload.ts (hunk 3 above)
Comment (asr2003) [reply src/bundle/upload.ts:37]: Testing it right now!

File: src/bundle/upload.ts [hunk 4]
```diff
@@ -301,223 +302,248 @@ async function uploadBundleToCapgoCloud(supabase: SupabaseType, appid: string, b
...
+      }
     }
   }
-}
 
-export async function uploadBundle(preAppid: string, options: Options, shouldExit = true) {
-  p.intro(`Uploading`)
-  const pm = getPMAndCommand()
-  await checkLatest()
+  export async function uploadBundle(preAppid: string, options: Options, shouldExit = true) {
```
Comment (ezhil56x) [inline src/bundle/upload.ts:414]: @asr2003 
Before committing, I would recommend you to run `npm run lint`

Comment (asr2003) [reply src/bundle/upload.ts:414]: Thanks for the suggestion @ezhil56x

**Comment by asr2003** [comment]
@riderx I am very close to solution but you finally fixed it :)

**Comment by asr2003** [comment]
@riderx Would I proceed migrating to Camera-V2 and wait for neo for reviewing or testing?

**Pull request closed without merging**